src/Microsoft.AspNet.OData/Formatter/Deserialization/ODataDeserializerContext.cs

```diff
@@ -36,7 +36,7 @@ public HttpRequestMessage Request
 
                 // We add this setting via CompatibilityOptions
                 CompatibilityOptions options = webApiRequestMessage != null ? webApiRequestMessage.Configuration.GetCompatibilityOptions() : CompatibilityOptions.None;
-                DisableCaseInsensitiveRequestPropertyBinding = options.HasFlag(CompatibilityOptions.DisableCaseInsensitiveRequestPropertyBinding) ? true : false;
```

Comment:
`HasFlag` will box and therefore allocate - instead of requiring bit checks for options I created the `HasOption` extension method instead.